### PR TITLE
TSK-1658: Add notIn-filter for the custom fields in the TaskQuery

### PR DESCRIPTION
**Pull request by Tristan2357**
Adapted the TaskQueryMapper
Adapted the TaskQuery
Expanded the (Rest) API

---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [x] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [x] Commit message format → TSK-XXX: Your commit message.
- [x] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [x] SonarCloud analysis meets our standards
- [x] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

**Comment by Tristan2357** [comment]
https://sonarcloud.io/dashboard?branch=TSK-1658&id=Tristan2357_taskana 
-> Fails, because i do not know how to properly test the Db2 versions of the queries. 
Also the about 30 Code Smells are all because of duplication of literal Strings, which i did not fix, because it would decrease the readability.

File: lib/taskana-core/src/test/java/acceptance/ArchitectureTest.java [hunk 1]
```diff
@@ -278,7 +279,8 @@ public void check(JavaClass javaClass, ConditionEvents events) {
         for (JavaMethod method : javaClass.getAllMethods()) {
           List<String> sqlStrings = getSqlStringsFromMethod.apply(method);
 
-          if (sqlStrings.isEmpty()) {
+          if (sqlStrings.isEmpty()
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/ArchitectureTest.java:282]: this "breaks" the architechture test. Let's have a look how we can fix that ;)

File: lib/taskana-core/src/test/java/acceptance/objectreference/QueryObjectReferencesWithPaginationAccTest.java [hunk 2]
```diff
@@ -11,25 +11,21 @@
...
+class QueryObjectReferencesWithPaginationAccTest extends AbstractAccTest {
 
-  private TaskService taskService;
-  private TaskQuery taskQuery;
   private ObjectReferenceQuery objRefQuery;
 
-  QueryObjectreferencesWithPaginationAccTest() {
+  QueryObjectReferencesWithPaginationAccTest() {
     super();
   }
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/objectreference/QueryObjectReferencesWithPaginationAccTest.java:24]: delete

File: lib/taskana-core/src/test/java/acceptance/task/QueryTasksAccTest.java [hunk 3]
```diff
@@ -163,6 +164,20 @@ void testQueryForOwnerLike() {
     }
   }
 
+  @WithAccessId(user = "admin")
+  @Test
+  void should_ReturnCorrectResults_When_QueryingForDescriptionAndPriority() {
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/QueryTasksAccTest.java:169]: split tests :)

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskQuerySqlProvider.java [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+        + "AND t.WORKBASKET_ID IN ("
+        + "SELECT WID from ("
+        + "SELECT WORKBASKET_ID "
+        + "as WID, MAX(PERM_READ::int) "
+        + "as MAX_READ FROM WORKBASKET_ACCESS_LIST "
+        + "AS s where ACCESS_ID IN "
+        + "(<foreach item='item' collection='accessIdIn' separator=',' >#{item}</foreach>) "
+        + "group by WORKBASKET_ID) "
+        + "AS f where max_read = 1) "
+        + "</if> "
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskQuerySqlProvider.java:66]: let's extract this to a method. A `checkForAuthorization` is more readable than this :)

File: lib/taskana-core/src/test/java/acceptance/task/QueryTasksAccTest.java (hunk 3 above)
Comment (mustaphazorgati) [reply lib/taskana-core/src/test/java/acceptance/task/QueryTasksAccTest.java:169]: still not split :)

File: lib/taskana-core/src/test/java/acceptance/task/QueryTasksAccTest.java [hunk 5]
```diff
@@ -297,23 +312,60 @@ void testQueryForExternalId() {
...
+            Pair.of(TaskCustomField.CUSTOM_7, new String[] {"%"}),
+            Pair.of(TaskCustomField.CUSTOM_8, new String[] {"%"}),
+            Pair.of(TaskCustomField.CUSTOM_9, new String[] {"%"}),
+            Pair.of(TaskCustomField.CUSTOM_10, new String[] {"custom10"}),
+            Pair.of(TaskCustomField.CUSTOM_11, new String[] {"custom11"}),
+            Pair.of(TaskCustomField.CUSTOM_12, new String[] {"custom12"}),
+            Pair.of(TaskCustomField.CUSTOM_13, new String[] {"custom13"}),
+            Pair.of(TaskCustomField.CUSTOM_14, new String[] {"abc"}),
+            Pair.of(TaskCustomField.CUSTOM_15, new String[] {"custom15"}),
+            Pair.of(TaskCustomField.CUSTOM_16, new String[] {"custom16"}));
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/QueryTasksAccTest.java:353]: Well done! 
I am aware that each result has only two hits. But I prefer a Triplet where the expected amount is viewable here.

Reason: If for some reason some test data gets added the number's will not match anymore

**Review by mustaphazorgati — APPROVED** [review]

**Pull request merged**